File: Binary_Search.py
def binary_search(arr, key):
    low = 0
    high = len(arr) - 1
    
    while low <= high:
        mid = (low + high) // 2
        if arr[mid] == key:
            return mid
        elif arr[mid] < key:
            low = mid + 1
        else:
            high = mid - 1
    return -1

n = int(input("Enter number of elements: "))
arr = []

print(f"Enter {n} sorted elements:")
for _ in range(n):
    arr.append(int(input()))

key = int(input("Enter element to search: "))

index = binary_search(arr, key)

if index != -1:
    print(f"Element found at position {index+1} (index {index})")
else:
    print("Element not found.")

File: Binary_Search.c
#include <stdio.h>

int main() {
    int n, i, key, low, high, mid;
    printf("Enter number of elements: ");
    scanf("%d", &n);
    
    int arr[n];
    printf("Enter %d sorted elements:\n", n);
    for (i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    
    printf("Enter element to search: ");
    scanf("%d", &key);
    
    low = 0;
    high = n - 1;
    
    while (low <= high) {
        mid = (low + high) / 2;
        
        if (arr[mid] == key) {
            printf("Element found at position %d (index %d)\n", mid+1, mid);
            return 0;
        } else if (arr[mid] < key) {
            low = mid + 1;
        } else {
            high = mid - 1;
        }
    }
    
    printf("Element not found.\n");
    return 0;
}
